### PR TITLE
Cleaned up unit tests

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#8246

*Description of changes:*
- updated vue test utils package
- deleted unused dev package
- changed most unit tests to async
- lint cleanup (in spec files only)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "build": "node build/build.js"
   },
   "dependencies": {
+    "@mdi/font": "^5.9.55",
```
Comment (severinbeauvais) [inline client/package.json:19]: This should be a regular dependency (not a dev-only dependency).

File: client/package.json [hunk 2]
```diff
@@ -24,16 +25,14 @@
     "vue-fragment": "^1.5.1",
     "vue-router": "^3.0.7",
     "vue-shortkey": "^3.1.4",
-    "vue-test-utils": "^1.0.0-beta.11",
```
Comment (severinbeauvais) [inline client/package.json:27]: This is not needed in a Prod build, so it is a dev-only dependency.

Comment (severinbeauvais) [inline client/package.json:27]: This would be a dev-only change, but I couldn't find anywhere this was used so I deleted it.

**Review by cameron-freshworks — APPROVED** [review]
Whoa! Looks good

**Pull request merged**